AppCheckCore/Sources/AppAttestProvider/GACAppAttestProvider.m

```diff
@@ -209,38 +203,56 @@ - (void)getTokenWithCompletion:(void (^)(GACAppCheckToken *_Nullable, NSError *_
       });
 }
 
-- (FBLPromise<GACAppCheckToken *> *)getToken {
-  return [FBLPromise onQueue:self.queue
-                          do:^id _Nullable {
-                            if (self.ongoingGetTokenOperation == nil) {
-                              // Kick off a new handshake sequence only when there is not an ongoing
-                              // handshake to avoid race conditions.
-                              self.ongoingGetTokenOperation =
-                                  [self createGetTokenSequenceWithBackoffPromise]
-
-                                      // Release the ongoing operation promise on completion.
-                                      .then(^GACAppCheckToken *(GACAppCheckToken *token) {
-                                        self.ongoingGetTokenOperation = nil;
-                                        return token;
-                                      })
-                                      .recover(^NSError *(NSError *error) {
-                                        self.ongoingGetTokenOperation = nil;
```

Comment:
Previously, `self.ongoingGetTokenOperation` was read on the `self.queue` dispatch queue but mutated on the main thread in `.then(...)` and `.recover(...)` (the main thread is the default when a queue is not specified) -- this could potentially result in a synchronization issue.